### PR TITLE
Refactor contribution tabs to have separate queries

### PR DESCRIPTION
**Pull request by kntsoriano**
https://github.com/exercism/v3-project-management/issues/199

### CODE REVIEW

File: app/javascript/components/profile/contributions-list/MaintainingContributionsList.tsx [hunk 1]
```diff
@@ -1,25 +1,68 @@
...
+type PaginatedResult = {
+  results: readonly ContributionProps[]
+  meta: {
+    currentPage: number
+    totalCount: number
+    totalPages: number
+  }
+}
+
+const DEFAULT_ERROR = new Error('Unable to load contributions')
```
Comment (ErikSchierboom) [inline app/javascript/components/profile/contributions-list/MaintainingContributionsList.tsx]: Maybe make this specific:

```suggestion
const DEFAULT_ERROR = new Error('Unable to load maintaining contributions')
```

File: app/javascript/components/profile/contributions-list/BuildingContributionsList.tsx [hunk 2]
```diff
@@ -1,25 +1,68 @@
...
+type PaginatedResult = {
+  results: readonly ContributionProps[]
+  meta: {
+    currentPage: number
+    totalCount: number
+    totalPages: number
+  }
+}
+
+const DEFAULT_ERROR = new Error('Unable to load contributions')
```
Comment (ErikSchierboom) [inline app/javascript/components/profile/contributions-list/BuildingContributionsList.tsx]: Maybe make this specific?

```suggestion
const DEFAULT_ERROR = new Error('Unable to load building contributions')
```

File: app/javascript/components/profile/contributions-list/AuthoringContributionsList.tsx [hunk 3]
```diff
@@ -1,26 +1,75 @@
...
+type PaginatedResult = {
+  results: readonly ExerciseAuthorship[]
+  meta: {
+    currentPage: number
+    totalCount: number
+    totalPages: number
+  }
+}
+
+const DEFAULT_ERROR = new Error('Unable to load contributions')
```
Comment (ErikSchierboom) [inline app/javascript/components/profile/contributions-list/AuthoringContributionsList.tsx]: Maybe make this specific?

```suggestion
const DEFAULT_ERROR = new Error('Unable to load authoring contributions')
```

File: app/javascript/components/profile/contributions-list/BuildingContributionsList.tsx [hunk 4]
```diff
@@ -1,25 +1,68 @@
...
+import { usePaginatedRequestQuery, Request } from '../../../hooks/request-query'
+
+type PaginatedResult = {
+  results: readonly ContributionProps[]
+  meta: {
+    currentPage: number
+    totalCount: number
+    totalPages: number
+  }
+}
```
Comment (ErikSchierboom) [inline app/javascript/components/profile/contributions-list/BuildingContributionsList.tsx:18]: I see this type of structure defined quite a bit. Would it make sense to introduce a `PaginatedResult<T>` type?

**Review by ErikSchierboom — APPROVED** [review]

**Comment by iHiD** [comment]
@kntsoriano I think this is loads better. You agree? 

I'll merge it now. Worth having a think about Erik's pagination suggestion too I think as we move into the refactoring phase soon :)

**Pull request merged**